docker.go

```diff
@@ -658,13 +676,27 @@ func (c *DockerContainer) StartLogProducer(ctx context.Context) error {
 						since = fmt.Sprintf("%d.%09d", now.Unix(), int64(now.Nanosecond()))
 						goto BEGIN
 					}
-					if errors.Is(err, context.DeadlineExceeded) {
-						// Probably safe to continue here
+					if errors.Is(err, context.DeadlineExceeded) || errors.Is(err, context.Canceled) {
+						// If the outer context is done, loop will exit in the next iteration.
 						continue
 					}
-					_, _ = fmt.Fprintf(os.Stderr, "container log error: %+v. %s", err, logStoppedForOutOfSyncMessage)
+
+					select {
+					case <-ctx.Done():
+						return
+					default:
+						c.logger.Printf("read log header: %+v. %s", err, logRestartedForOutOfSyncMessage)
+					}
```

Comment:
You'd have to excuse me for my previous comments. While this is totally OK, the `select` statement isn't really necessary here. There's a very slim chance of hitting the `case <-ctx.Done()` branch here so we might as well remove the whole `select` statement here.

Reply:
Do you want the default case to be moved to the second select, or simply remove it completely?

Reply:
We can basically do this:

```suggestion
					c.logger.Printf("read log header: %+v. %s", err, logRestartedForOutOfSyncMessage)
```

Reply:
Ok cool! we are going to roll it back to the initial state where the print was out of the select 😅 

In any case, thanks for your assistance in this part of the code, some time you find it difficult to control all of them